api/net/Areas/Subscriber/Controllers/ProductController.cs

```diff
@@ -0,0 +1,147 @@
+[ProducesResponseType(typeof(ErrorResponseModel), (int)HttpStatusCode.Forbidden)]
+public class ProductController : ControllerBase
+{
+    #region Variables
+    private readonly IProductService _productService;
+    private readonly IUserService _userService;
+    #endregion
+
+    #region Constructors
+    /// <summary>
+    /// Creates a new instance of a ProductController object, initializes with specified parameters.
+    /// </summary>
+    /// <param name="productService"></param>
+    /// <param name="userService"></param>
+    public ProductController(
+        IProductService productService,
+        IUserService userService)
+    {
+        _productService = productService;
+        _userService = userService;
+    }
+    #endregion
+
+    #region Endpoints
+
+    /// <summary>
+    /// Find all products - returns subscription status for current user.
+    /// </summary>
+    /// <returns></returns>
+    [HttpGet("")]
+    [Produces(MediaTypeNames.Application.Json)]
+    [ProducesResponseType(typeof(IEnumerable<ProductModel>), (int)HttpStatusCode.OK)]
+    [SwaggerOperation(Tags = new[] { "Product" })]
+    public IActionResult FindMyProducts()
+    {
+        var username = User.GetUsername() ?? throw new NotAuthorizedException("Username is missing");
+        var user = _userService.FindByUsername(username) ?? throw new NotAuthorizedException("User does not exist");
+        return new JsonResult(_productService.FindProductsByUser(user.Id).Select(ds => new ProductModel(ds)));
+    }
+
+    /// <summary>
+    /// Find all "available" products.
+    /// </summary>
+    /// <returns></returns>
+    [HttpGet]
+    [Produces(MediaTypeNames.Application.Json)]
+    [ProducesResponseType(typeof(IEnumerable<ProductModel>), (int)HttpStatusCode.OK)]
+    [SwaggerOperation(Tags = new[] { "Product" })]
+    public IActionResult GetProducts()
+    {
+        return new JsonResult(_productService.FindAll().Select(ds => new ProductModel(ds)));
+    }
+
+    /// <summary>
+    /// Add product.
+    /// </summary>
+    /// <param name="model"></param>
+    /// <returns></returns>
+    [HttpPost]
+    [Produces(MediaTypeNames.Application.Json)]
+    [ProducesResponseType(typeof(ProductModel), (int)HttpStatusCode.Created)]
+    [ProducesResponseType(typeof(ErrorResponseModel), (int)HttpStatusCode.BadRequest)]
+    [SwaggerOperation(Tags = new[] { "Product" })]
+    public IActionResult SubscribeCurrentUser(ProductModel model)
+    {
+        var username = User.GetUsername() ?? throw new NotAuthorizedException("Username is missing");
+        var user = _userService.FindByUsername(username) ?? throw new NotAuthorizedException("User does not exist");
+        var result = _productService.FindById(model.Id) ?? throw new NoContentException("Product does not exist");
+
+        // If the current user is not already subscribed, add them to the list of subscribers but not in a subscribed state.
+        if (!result.Subscribers.Exists(s => s.Id == user.Id))
```

Comment:
how are you handling a request to unsubscribe?

Reply:
added the plumbing for this.  I have a TODO on my desk for the plumbing for `Notification` and `Evening Overview`